scipy/stats/tests/test_axis_nan_policy.py

```diff
@@ -30,6 +30,7 @@
     (stats.skew, tuple(), dict(), 1, 1, False, lambda x: (x,)),
     (stats.kstat, tuple(), dict(), 1, 1, False, lambda x: (x,)),
     (stats.kstatvar, tuple(), dict(), 1, 1, False, lambda x: (x,)),
+    (stats.moment, tuple(), dict(), 1, 1, False, lambda x: (x,)),
```

Comment:
Could you try adding an array moment argument test case here? I tried once and it complained a bit, but hopefully it was my mistake.

Reply:
When I add the test case `(stats.moment, tuple(), dict(moment=[1, 2]), 1, 1, False, lambda x: (x,))`, it gives me four errors:

```
```

Reply:
When I add the test case `(stats.moment, tuple(), dict(moment=[1, 2]), 1, 1, False, lambda x: (x,))`, it gives me four errors:

```
FAILED scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-propagate-moment-args12-kwds12-1-1-False-<lambda>] - ValueError: setting an array element with a sequence.
FAILED scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-moment-args12-kwds12-1-1-False-<lambda>] - ValueError: setting an array element with a sequence.
FAILED scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-raise-moment-args12-kwds12-1-1-False-<lambda>] - ValueError: setting an array element with a sequence.
FAILED scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_axis_is_None[mixed-propagate-moment-args12-kwds12-1-1-False-<lambda>] - AssertionError: 
FAILED scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_axis_is_None[all_nans-propagate-moment-args12-kwds12-1-1-False-<lambda>] - AssertionError:
```

All of these are either because the test tried to write an array to a scalar element of an array or because of slight shape mispatch. I verified and the output shapes is correct for each of the failing test. So, I don't think there are any actual failures.

<details>

<summary>Full test log</summary>


```
ninja: Entering directory `build'
[2/2] Generating scipy/generate-config with a custom command
Build OK
Installing, see meson-install.log...
Installation OK
Running tests for scipy version:1.9.0.dev0+1979.39c9eb2, installed at:/home/tirthasheshpatel/tensor_libs/scipy/build-install/lib/python3.10/site-packages/scipy
=========================================================================================== test session starts ===========================================================================================
platform linux -- Python 3.10.4, pytest-7.0.1, pluggy-1.0.0
rootdir: /home/tirthasheshpatel/tensor_libs/scipy, configfile: pytest.ini
plugins: cov-3.0.0, xdist-2.5.0, forked-1.4.0
gw0 [273] / gw1 [273] / gw2 [273] / gw3 [273] / gw4 [273] / gw5 [273] / gw6 [273] / gw7 [273]
...........F......F....................................................................F.F..........F.............................................................................................. [ 71%]
..............................................................................                                                                                                                      [100%]
================================================================================================ FAILURES =================================================================================================
________________________________________________________ test_axis_nan_policy_fast[mixed-1-propagate-hypotest12-args12-kwds12-1-1-False-<lambda>] _________________________________________________________
[gw4] linux -- Python 3.10.4 /home/tirthasheshpatel/tensor_libs/virtualenvs/scipy-dev/bin/python
TypeError: only size-1 arrays can be converted to Python scalars

The above exception was the direct cause of the following exception:
scipy/stats/tests/test_axis_nan_policy.py:157: in test_axis_nan_policy_fast
    _axis_nan_policy_test(hypotest, args, kwds, n_samples, n_outputs, paired,
        args       = ()
        axis       = 1
        data_generator = 'mixed'
        hypotest   = functools.partial(<function moment at 0x7ff0a0b76c20>, moment=[1, 2, 3])
        kwds       = {}
        n_outputs  = 1
        n_samples  = 1
        nan_policy = 'propagate'
        paired     = False
        unpacker   = <function <lambda> at 0x7ff0a08aea70>
scipy/stats/tests/test_axis_nan_policy.py:267: in _axis_nan_policy_test
    statistics[i] = res1d[0]
E   ValueError: setting an array element with a sequence.
        _          = 0.0
        args       = ()
        axis       = 1
        data       = [array([[[       nan, 0.56223184, 0.4974227 ,        nan,        nan,
                nan],
        [       nan, 0.040...an,
                nan],
        [       nan, 0.51217366, 0.3946598 ,        nan,        nan,
                nan]]])]
        data1d     = [array([0.56223184, 0.04051071, 0.0660825 , 0.96606208, 0.92910422,
       0.51275872, 0.757951  , 0.06669001, 0.93101...1, 0.7320062 , 0.02836537, 0.61437325, 0.44275283,
       0.34430998, 0.43029873, 0.01599173, 0.84131728, 0.71921977])]
        data_b     = [array([[[       nan,        nan,        nan,        nan,        nan,
                nan,        nan,        nan,    ...      nan,        nan,        nan, 0.42373744,
                nan,        nan,        nan,        nan,        nan]]])]
        data_gen_kwds = {'axis': 1, 'n_repetitions': 3, 'n_samples': 1, 'paired': False, ...}
        data_generator = 'mixed'
        hypotest   = functools.partial(<function moment at 0x7ff0a0b76c20>, moment=[1, 2, 3])
        i          = (0, 1)
        inherent_size = 6
        kwds       = {}
        n_outputs  = 1
        n_repetitions = 3
        n_samples  = 1
        nan_policy = 'propagate'
        output_shape = [3, 6]
        paired     = False
        pvalues    = array([[0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0.]])
        res1d      = (array([ 0.        ,  0.10115234, -0.00467782]),)
        res1db     = (array([ 0.        ,  0.10115234, -0.00467782]),)
        rng        = Generator(PCG64) at 0x7FF0A0324AC0
        statistics = array([[nan,  0.,  0.,  0.,  0.,  0.],
       [ 0.,  0.,  0.,  0.,  0.,  0.],
       [ 0.,  0.,  0.,  0.,  0.,  0.]])
        unpacker   = <function <lambda> at 0x7ff0a08aea70>
___________________________________________________________ test_axis_nan_policy_fast[mixed-1-omit-hypotest12-args12-kwds12-1-1-False-<lambda>] ___________________________________________________________
[gw1] linux -- Python 3.10.4 /home/tirthasheshpatel/tensor_libs/virtualenvs/scipy-dev/bin/python
TypeError: only size-1 arrays can be converted to Python scalars

The above exception was the direct cause of the following exception:
scipy/stats/tests/test_axis_nan_policy.py:157: in test_axis_nan_policy_fast
    _axis_nan_policy_test(hypotest, args, kwds, n_samples, n_outputs, paired,
        args       = ()
        axis       = 1
        data_generator = 'mixed'
        hypotest   = functools.partial(<function moment at 0x7f6e435425f0>, moment=[1, 2, 3])
        kwds       = {}
        n_outputs  = 1
        n_samples  = 1
        nan_policy = 'omit'
        paired     = False
        unpacker   = <function <lambda> at 0x7f6e4327e7a0>
scipy/stats/tests/test_axis_nan_policy.py:267: in _axis_nan_policy_test
    statistics[i] = res1d[0]
E   ValueError: setting an array element with a sequence.
        _          = 0.0
        args       = ()
        axis       = 1
        data       = [array([[[       nan, 0.56223184, 0.4974227 ,        nan,        nan,
                nan],
        [       nan, 0.040...an,
                nan],
        [       nan, 0.51217366, 0.3946598 ,        nan,        nan,
                nan]]])]
        data1d     = [array([       nan,        nan,        nan,        nan,        nan,
              nan,        nan,        nan,        ...n, 0.38892142,        nan,        nan,        nan,
              nan,        nan,        nan,        nan,        nan])]
        data_b     = [array([[[       nan,        nan,        nan,        nan,        nan,
                nan,        nan,        nan,    ...      nan,        nan,        nan, 0.42373744,
                nan,        nan,        nan,        nan,        nan]]])]
        data_gen_kwds = {'axis': 1, 'n_repetitions': 3, 'n_samples': 1, 'paired': False, ...}
        data_generator = 'mixed'
        hypotest   = functools.partial(<function moment at 0x7f6e435425f0>, moment=[1, 2, 3])
        i          = (0, 0)
        inherent_size = 6
        kwds       = {}
        n_outputs  = 1
        n_repetitions = 3
        n_samples  = 1
        nan_policy = 'omit'
        output_shape = [3, 6]
        paired     = False
        pvalues    = array([[0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0.]])
        res1d      = (array([0., 0., 0.]),)
        rng        = Generator(PCG64) at 0x7F6E42D149E0
        statistics = array([[0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0.]])
        sup        = <numpy.testing._private.utils.suppress_warnings object at 0x7f6e4330b1c0>
        unpacker   = <function <lambda> at 0x7f6e4327e7a0>
_____________________________________________________ test_axis_nan_policy_axis_is_None[mixed-propagate-hypotest12-args12-kwds12-1-1-False-<lambda>] ______________________________________________________
[gw1] linux -- Python 3.10.4 /home/tirthasheshpatel/tensor_libs/virtualenvs/scipy-dev/bin/python
scipy/stats/tests/test_axis_nan_policy.py:368: in test_axis_nan_policy_axis_is_None
    assert_equal(res1db, res1da)
E   AssertionError: 
E   Arrays are not equal
E   
E   (shapes (1, 3), (1,) mismatch)
E    x: array([[nan, nan, nan]])
E    y: array([nan])
        args       = ()
        data       = [array([[0.63696169, 0.26978671, 0.04097352, 0.01652764, 0.81327024,
        0.91275558, 0.60663578, 0.72949656, 0.543...        nan, 0.13509651,        nan, 0.52535432,
        0.31024188, 0.48583536,        nan,        nan, 0.3577952 ]])]
        data_generator = 'mixed'
        data_raveled = [array([0.63696169, 0.26978671, 0.04097352, 0.01652764, 0.81327024,
       0.91275558, 0.60663578, 0.72949656, 0.54362...3,        nan, 0.13509651,        nan, 0.52535432,
       0.31024188, 0.48583536,        nan,        nan, 0.3577952 ])]
        ea_str     = None
        eb_str     = None
        ec_str     = None
        hypotest   = functools.partial(<function moment at 0x7f6e435425f0>, moment=[1, 2, 3])
        kwds       = {}
        mask       = array([[False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, Fals...se, False,  True, False,
        False, False,  True, False,  True, False, False, False,  True,
         True, False]])
        masks      = [array([[False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, Fal...e, False,  True, False,
        False, False,  True, False,  True, False, False, False,  True,
         True, False]])]
        n_outputs  = 1
        n_samples  = 1
        nan_policy = 'propagate'
        paired     = False
        res1da     = array([nan])
        res1db     = (array([nan, nan, nan]),)
        res1dc     = (array([nan, nan, nan]),)
        rng        = Generator(PCG64) at 0x7F6E42EBF920
        sample     = array([[0.63696169, 0.26978671, 0.04097352, 0.01652764, 0.81327024,
        0.91275558, 0.60663578, 0.72949656, 0.5436...,        nan, 0.13509651,        nan, 0.52535432,
        0.31024188, 0.48583536,        nan,        nan, 0.3577952 ]])
        unpacker   = <function <lambda> at 0x7f6e4327e7a0>
__________________________________________________________ test_axis_nan_policy_fast[mixed-1-raise-hypotest12-args12-kwds12-1-1-False-<lambda>] ___________________________________________________________
[gw6] linux -- Python 3.10.4 /home/tirthasheshpatel/tensor_libs/virtualenvs/scipy-dev/bin/python
TypeError: only size-1 arrays can be converted to Python scalars

The above exception was the direct cause of the following exception:
scipy/stats/tests/test_axis_nan_policy.py:157: in test_axis_nan_policy_fast
    _axis_nan_policy_test(hypotest, args, kwds, n_samples, n_outputs, paired,
        args       = ()
        axis       = 1
        data_generator = 'mixed'
        hypotest   = functools.partial(<function moment at 0x7f6716b42c20>, moment=[1, 2, 3])
        kwds       = {}
        n_outputs  = 1
        n_samples  = 1
        nan_policy = 'raise'
        paired     = False
        unpacker   = <function <lambda> at 0x7f671687ea70>
scipy/stats/tests/test_axis_nan_policy.py:267: in _axis_nan_policy_test
    statistics[i] = res1d[0]
E   ValueError: setting an array element with a sequence.
        _          = 0.0
        args       = ()
        axis       = 1
        data       = [array([[[       nan, 0.56223184, 0.4974227 ,        nan,        nan,
                nan],
        [       nan, 0.040...an,
                nan],
        [       nan, 0.51217366, 0.3946598 ,        nan,        nan,
                nan]]])]
        data1d     = [array([0.56223184, 0.04051071, 0.0660825 , 0.96606208, 0.92910422,
       0.51275872, 0.757951  , 0.06669001, 0.93101...1, 0.7320062 , 0.02836537, 0.61437325, 0.44275283,
       0.34430998, 0.43029873, 0.01599173, 0.84131728, 0.71921977])]
        data_b     = [array([[[       nan,        nan,        nan,        nan,        nan,
                nan,        nan,        nan,    ...      nan,        nan,        nan, 0.42373744,
                nan,        nan,        nan,        nan,        nan]]])]
        data_gen_kwds = {'axis': 1, 'n_repetitions': 3, 'n_samples': 1, 'paired': False, ...}
        data_generator = 'mixed'
        hypotest   = functools.partial(<function moment at 0x7f6716b42c20>, moment=[1, 2, 3])
        i          = (0, 1)
        inherent_size = 6
        kwds       = {}
        n_outputs  = 1
        n_repetitions = 3
        n_samples  = 1
        nan_policy = 'raise'
        output_shape = [3, 6]
        paired     = False
        pvalues    = array([[0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0.]])
        res1d      = (array([ 0.        ,  0.10115234, -0.00467782]),)
        res1db     = (array([ 0.        ,  0.10115234, -0.00467782]),)
        rng        = Generator(PCG64) at 0x7F67162F8820
        statistics = array([[nan,  0.,  0.,  0.,  0.,  0.],
       [ 0.,  0.,  0.,  0.,  0.,  0.],
       [ 0.,  0.,  0.,  0.,  0.,  0.]])
        unpacker   = <function <lambda> at 0x7f671687ea70>
____________________________________________________ test_axis_nan_policy_axis_is_None[all_nans-propagate-hypotest12-args12-kwds12-1-1-False-<lambda>] ____________________________________________________
[gw3] linux -- Python 3.10.4 /home/tirthasheshpatel/tensor_libs/virtualenvs/scipy-dev/bin/python
scipy/stats/tests/test_axis_nan_policy.py:368: in test_axis_nan_policy_axis_is_None
    assert_equal(res1db, res1da)
E   AssertionError: 
E   Arrays are not equal
E   
E   (shapes (1, 3), (1,) mismatch)
E    x: array([[nan, nan, nan]])
E    y: array([nan])
        args       = ()
        data       = [array([[nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan, nan, nan]])]
        data_generator = 'all_nans'
        data_raveled = [array([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
       nan])]
        ea_str     = None
        eb_str     = None
        ec_str     = None
        hypotest   = functools.partial(<function moment at 0x7fc42d66ec20>, moment=[1, 2, 3])
        kwds       = {}
        n_outputs  = 1
        n_samples  = 1
        nan_policy = 'propagate'
        paired     = False
        res1da     = array([nan])
        res1db     = (array([nan, nan, nan]),)
        res1dc     = (array([nan, nan, nan]),)
        rng        = Generator(PCG64) at 0x7FC42CE24820
        unpacker   = <function <lambda> at 0x7fc42d3aaa70>
========================================================================================= short test summary info =========================================================================================
FAILED scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-propagate-hypotest12-args12-kwds12-1-1-False-<lambda>] - ValueError: setting an array element with a sequence.
FAILED scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-hypotest12-args12-kwds12-1-1-False-<lambda>] - ValueError: setting an array element with a sequence.
FAILED scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_axis_is_None[mixed-propagate-hypotest12-args12-kwds12-1-1-False-<lambda>] - AssertionError: 
FAILED scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-raise-hypotest12-args12-kwds12-1-1-False-<lambda>] - ValueError: setting an array element with a sequence.
FAILED scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_axis_is_None[all_nans-propagate-hypotest12-args12-kwds12-1-1-False-<lambda>] - AssertionError: 
====================================================================================== 5 failed, 268 passed in 4.42s ======================================================================================
(scipy-dev) tirthasheshpatel@tirthap:~/tensor_libs/scipy$ SCIPY_XSLOW=1 OMP_NUM_THREADS=8 python dev.py -t scipy/stats/tests/test_axis_nan_policy.py -j8
ninja: Entering directory `build'
[2/2] Generating scipy/generate-config with a custom command
Build OK
Installing, see meson-install.log...
Installation OK
Running tests for scipy version:1.9.0.dev0+1979.39c9eb2, installed at:/home/tirthasheshpatel/tensor_libs/scipy/build-install/lib/python3.10/site-packages/scipy
=========================================================================================== test session starts ===========================================================================================
platform linux -- Python 3.10.4, pytest-7.0.1, pluggy-1.0.0
rootdir: /home/tirthasheshpatel/tensor_libs/scipy, configfile: pytest.ini
plugins: cov-3.0.0, xdist-2.5.0, forked-1.4.0
gw0 [273] / gw1 [273] / gw2 [273] / gw3 [273] / gw4 [273] / gw5 [273] / gw6 [273] / gw7 [273]
.......................................F........................................................................................................................................................... [ 71%]
..............................................................................                                                                                                                      [100%]
================================================================================================ FAILURES =================================================================================================
___________________________________________________________ test_axis_nan_policy_fast[mixed-1-omit-hypotest12-args12-kwds12-1-1-False-<lambda>] ___________________________________________________________
[gw1] linux -- Python 3.10.4 /home/tirthasheshpatel/tensor_libs/virtualenvs/scipy-dev/bin/python
scipy/stats/tests/test_axis_nan_policy.py:157: in test_axis_nan_policy_fast
    _axis_nan_policy_test(hypotest, args, kwds, n_samples, n_outputs, paired,
        args       = ()
        axis       = 1
        data_generator = 'mixed'
        hypotest   = functools.partial(<function moment at 0x7f3dd411ec20>, moment=[1])
        kwds       = {}
        n_outputs  = 1
        n_samples  = 1
        nan_policy = 'omit'
        paired     = False
        unpacker   = <function <lambda> at 0x7f3dd3e5add0>
scipy/stats/tests/test_axis_nan_policy.py:286: in _axis_nan_policy_test
    assert_equal(res[0], statistics)
E   AssertionError: 
E   Arrays are not equal
E   
E   (shapes (1, 3, 6), (3, 6) mismatch)
E    x: array([[[ 0.,  0.,  0.,  0.,  0., nan],
E           [ 0.,  0.,  0.,  0., nan,  0.],
E           [ 0.,  0.,  0.,  0., nan,  0.]]])
E    y: array([[ 0.,  0.,  0.,  0.,  0., nan],
E          [ 0.,  0.,  0.,  0., nan,  0.],
E          [ 0.,  0.,  0.,  0., nan,  0.]])
        _          = 0.0
        args       = ()
        axis       = 1
        data       = [array([[[       nan, 0.56223184, 0.4974227 ,        nan,        nan,
                nan],
        [       nan, 0.040...an,
                nan],
        [       nan, 0.51217366, 0.3946598 ,        nan,        nan,
                nan]]])]
        data1d     = [array([       nan, 0.42853961,        nan,        nan,        nan,
              nan,        nan,        nan,        ...n,        nan,        nan,        nan, 0.42373744,
              nan,        nan,        nan,        nan,        nan])]
        data_b     = [array([[[       nan,        nan,        nan,        nan,        nan,
                nan,        nan,        nan,    ...      nan,        nan,        nan, 0.42373744,
                nan,        nan,        nan,        nan,        nan]]])]
        data_gen_kwds = {'axis': 1, 'n_repetitions': 3, 'n_samples': 1, 'paired': False, ...}
        data_generator = 'mixed'
        hypotest   = functools.partial(<function moment at 0x7f3dd411ec20>, moment=[1])
        i          = (2, 5)
        inherent_size = 6
        kwds       = {}
        n_outputs  = 1
        n_repetitions = 3
        n_samples  = 1
        nan_policy = 'omit'
        output_shape = [3, 6]
        paired     = False
        pvalues    = array([[0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0.]])
        res        = (array([[[ 0.,  0.,  0.,  0.,  0., nan],
        [ 0.,  0.,  0.,  0., nan,  0.],
        [ 0.,  0.,  0.,  0., nan,  0.]]]),)
        res1d      = (array([0.]),)
        res1db     = (array([0.]),)
        rng        = Generator(PCG64) at 0x7F3DD38FCD60
        statistics = array([[ 0.,  0.,  0.,  0.,  0., nan],
       [ 0.,  0.,  0.,  0., nan,  0.],
       [ 0.,  0.,  0.,  0., nan,  0.]])
        sup        = <numpy.testing._private.utils.suppress_warnings object at 0x7f3dd3e4eef0>
        unpacker   = <function <lambda> at 0x7f3dd3e5add0>
========================================================================================= short test summary info =========================================================================================
FAILED scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-hypotest12-args12-kwds12-1-1-False-<lambda>] - AssertionError: 
====================================================================================== 1 failed, 272 passed in 4.21s ======================================================================================
(scipy-dev) tirthasheshpatel@tirthap:~/tensor_libs/scipy$ SCIPY_XSLOW=1 OMP_NUM_THREADS=8 python dev.py -t scipy/stats/tests/test_axis_nan_policy.py -j8
ninja: Entering directory `build'
[2/2] Generating scipy/generate-config with a custom command
Build OK
Installing, see meson-install.log...
Installation OK
Running tests for scipy version:1.9.0.dev0+1979.39c9eb2, installed at:/home/tirthasheshpatel/tensor_libs/scipy/build-install/lib/python3.10/site-packages/scipy
=========================================================================================== test session starts ===========================================================================================
platform linux -- Python 3.10.4, pytest-7.0.1, pluggy-1.0.0
rootdir: /home/tirthasheshpatel/tensor_libs/scipy, configfile: pytest.ini
plugins: cov-3.0.0, xdist-2.5.0, forked-1.4.0
gw0 [273] / gw1 [273] / gw2 [273] / gw3 [273] / gw4 [273] / gw5 [273] / gw6 [273] / gw7 [273]
.........F.....................F................................................F...F.F............................................................................................................ [ 71%]
..............................................................................                                                                                                                      [100%]
================================================================================================ FAILURES =================================================================================================
__________________________________________________________ test_axis_nan_policy_fast[mixed-1-propagate-moment-args12-kwds12-1-1-False-<lambda>] ___________________________________________________________
[gw3] linux -- Python 3.10.4 /home/tirthasheshpatel/tensor_libs/virtualenvs/scipy-dev/bin/python
TypeError: only size-1 arrays can be converted to Python scalars

The above exception was the direct cause of the following exception:
scipy/stats/tests/test_axis_nan_policy.py:155: in test_axis_nan_policy_fast
    _axis_nan_policy_test(hypotest, args, kwds, n_samples, n_outputs, paired,
        args       = ()
        axis       = 1
        data_generator = 'mixed'
        hypotest   = <function moment at 0x7f2e85c5ac20>
        kwds       = {'moment': [1, 2]}
        n_outputs  = 1
        n_samples  = 1
        nan_policy = 'propagate'
        paired     = False
        unpacker   = <function <lambda> at 0x7f2e8598ea70>
scipy/stats/tests/test_axis_nan_policy.py:265: in _axis_nan_policy_test
    statistics[i] = res1d[0]
E   ValueError: setting an array element with a sequence.
        _          = 0.0
        args       = ()
        axis       = 1
        data       = [array([[[       nan, 0.56223184, 0.4974227 ,        nan,        nan,
                nan],
        [       nan, 0.040...an,
                nan],
        [       nan, 0.51217366, 0.3946598 ,        nan,        nan,
                nan]]])]
        data1d     = [array([0.56223184, 0.04051071, 0.0660825 , 0.96606208, 0.92910422,
       0.51275872, 0.757951  , 0.06669001, 0.93101...1, 0.7320062 , 0.02836537, 0.61437325, 0.44275283,
       0.34430998, 0.43029873, 0.01599173, 0.84131728, 0.71921977])]
        data_b     = [array([[[       nan,        nan,        nan,        nan,        nan,
                nan,        nan,        nan,    ...      nan,        nan,        nan, 0.42373744,
                nan,        nan,        nan,        nan,        nan]]])]
        data_gen_kwds = {'axis': 1, 'n_repetitions': 3, 'n_samples': 1, 'paired': False, ...}
        data_generator = 'mixed'
        hypotest   = <function moment at 0x7f2e85c5ac20>
        i          = (0, 1)
        inherent_size = 6
        kwds       = {'moment': [1, 2]}
        n_outputs  = 1
        n_repetitions = 3
        n_samples  = 1
        nan_policy = 'propagate'
        output_shape = [3, 6]
        paired     = False
        pvalues    = array([[0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0.]])
        res1d      = (array([0.        , 0.10115234]),)
        res1db     = (array([0.        , 0.10115234]),)
        rng        = Generator(PCG64) at 0x7F2E853FD9A0
        statistics = array([[nan,  0.,  0.,  0.,  0.,  0.],
       [ 0.,  0.,  0.,  0.,  0.,  0.],
       [ 0.,  0.,  0.,  0.,  0.,  0.]])
        unpacker   = <function <lambda> at 0x7f2e8598ea70>
_____________________________________________________________ test_axis_nan_policy_fast[mixed-1-omit-moment-args12-kwds12-1-1-False-<lambda>] _____________________________________________________________
[gw1] linux -- Python 3.10.4 /home/tirthasheshpatel/tensor_libs/virtualenvs/scipy-dev/bin/python
TypeError: only size-1 arrays can be converted to Python scalars

The above exception was the direct cause of the following exception:
scipy/stats/tests/test_axis_nan_policy.py:155: in test_axis_nan_policy_fast
    _axis_nan_policy_test(hypotest, args, kwds, n_samples, n_outputs, paired,
        args       = ()
        axis       = 1
        data_generator = 'mixed'
        hypotest   = <function moment at 0x7f0ff9fc2c20>
        kwds       = {'moment': [1, 2]}
        n_outputs  = 1
        n_samples  = 1
        nan_policy = 'omit'
        paired     = False
        unpacker   = <function <lambda> at 0x7f0ff9cfadd0>
scipy/stats/tests/test_axis_nan_policy.py:265: in _axis_nan_policy_test
    statistics[i] = res1d[0]
E   ValueError: setting an array element with a sequence.
        _          = 0.0
        args       = ()
        axis       = 1
        data       = [array([[[       nan, 0.56223184, 0.4974227 ,        nan,        nan,
                nan],
        [       nan, 0.040...an,
                nan],
        [       nan, 0.51217366, 0.3946598 ,        nan,        nan,
                nan]]])]
        data1d     = [array([       nan,        nan,        nan,        nan,        nan,
              nan,        nan,        nan,        ...n, 0.38892142,        nan,        nan,        nan,
              nan,        nan,        nan,        nan,        nan])]
        data_b     = [array([[[       nan,        nan,        nan,        nan,        nan,
                nan,        nan,        nan,    ...      nan,        nan,        nan, 0.42373744,
                nan,        nan,        nan,        nan,        nan]]])]
        data_gen_kwds = {'axis': 1, 'n_repetitions': 3, 'n_samples': 1, 'paired': False, ...}
        data_generator = 'mixed'
        hypotest   = <function moment at 0x7f0ff9fc2c20>
        i          = (0, 0)
        inherent_size = 6
        kwds       = {'moment': [1, 2]}
        n_outputs  = 1
        n_repetitions = 3
        n_samples  = 1
        nan_policy = 'omit'
        output_shape = [3, 6]
        paired     = False
        pvalues    = array([[0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0.]])
        res1d      = (array([0., 0.]),)
        rng        = Generator(PCG64) at 0x7F0FF97898C0
        statistics = array([[0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0.]])
        sup        = <numpy.testing._private.utils.suppress_warnings object at 0x7f0ff9cf30a0>
        unpacker   = <function <lambda> at 0x7f0ff9cfadd0>
____________________________________________________________ test_axis_nan_policy_fast[mixed-1-raise-moment-args12-kwds12-1-1-False-<lambda>] _____________________________________________________________
[gw6] linux -- Python 3.10.4 /home/tirthasheshpatel/tensor_libs/virtualenvs/scipy-dev/bin/python
TypeError: only size-1 arrays can be converted to Python scalars

The above exception was the direct cause of the following exception:
scipy/stats/tests/test_axis_nan_policy.py:155: in test_axis_nan_policy_fast
    _axis_nan_policy_test(hypotest, args, kwds, n_samples, n_outputs, paired,
        args       = ()
        axis       = 1
        data_generator = 'mixed'
        hypotest   = <function moment at 0x7f79508bac20>
        kwds       = {'moment': [1, 2]}
        n_outputs  = 1
        n_samples  = 1
        nan_policy = 'raise'
        paired     = False
        unpacker   = <function <lambda> at 0x7f79505f2a70>
scipy/stats/tests/test_axis_nan_policy.py:265: in _axis_nan_policy_test
    statistics[i] = res1d[0]
E   ValueError: setting an array element with a sequence.
        _          = 0.0
        args       = ()
        axis       = 1
        data       = [array([[[       nan, 0.56223184, 0.4974227 ,        nan,        nan,
                nan],
        [       nan, 0.040...an,
                nan],
        [       nan, 0.51217366, 0.3946598 ,        nan,        nan,
                nan]]])]
        data1d     = [array([0.56223184, 0.04051071, 0.0660825 , 0.96606208, 0.92910422,
       0.51275872, 0.757951  , 0.06669001, 0.93101...1, 0.7320062 , 0.02836537, 0.61437325, 0.44275283,
       0.34430998, 0.43029873, 0.01599173, 0.84131728, 0.71921977])]
        data_b     = [array([[[       nan,        nan,        nan,        nan,        nan,
                nan,        nan,        nan,    ...      nan,        nan,        nan, 0.42373744,
                nan,        nan,        nan,        nan,        nan]]])]
        data_gen_kwds = {'axis': 1, 'n_repetitions': 3, 'n_samples': 1, 'paired': False, ...}
        data_generator = 'mixed'
        hypotest   = <function moment at 0x7f79508bac20>
        i          = (0, 1)
        inherent_size = 6
        kwds       = {'moment': [1, 2]}
        n_outputs  = 1
        n_repetitions = 3
        n_samples  = 1
        nan_policy = 'raise'
        output_shape = [3, 6]
        paired     = False
        pvalues    = array([[0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0.]])
        res1d      = (array([0.        , 0.10115234]),)
        res1db     = (array([0.        , 0.10115234]),)
        rng        = Generator(PCG64) at 0x7F79500617E0
        statistics = array([[nan,  0.,  0.,  0.,  0.,  0.],
       [ 0.,  0.,  0.,  0.,  0.,  0.],
       [ 0.,  0.,  0.,  0.,  0.,  0.]])
        unpacker   = <function <lambda> at 0x7f79505f2a70>
_______________________________________________________ test_axis_nan_policy_axis_is_None[mixed-propagate-moment-args12-kwds12-1-1-False-<lambda>] ________________________________________________________
[gw3] linux -- Python 3.10.4 /home/tirthasheshpatel/tensor_libs/virtualenvs/scipy-dev/bin/python
scipy/stats/tests/test_axis_nan_policy.py:366: in test_axis_nan_policy_axis_is_None
    assert_equal(res1db, res1da)
E   AssertionError: 
E   Arrays are not equal
E   
E   (shapes (1, 2), (1,) mismatch)
E    x: array([[nan, nan]])
E    y: array([nan])
        args       = ()
        data       = [array([[0.63696169, 0.26978671, 0.04097352, 0.01652764, 0.81327024,
        0.91275558, 0.60663578, 0.72949656, 0.543...        nan, 0.13509651,        nan, 0.52535432,
        0.31024188, 0.48583536,        nan,        nan, 0.3577952 ]])]
        data_generator = 'mixed'
        data_raveled = [array([0.63696169, 0.26978671, 0.04097352, 0.01652764, 0.81327024,
       0.91275558, 0.60663578, 0.72949656, 0.54362...3,        nan, 0.13509651,        nan, 0.52535432,
       0.31024188, 0.48583536,        nan,        nan, 0.3577952 ])]
        ea_str     = None
        eb_str     = None
        ec_str     = None
        hypotest   = <function moment at 0x7f2e85c5ac20>
        kwds       = {'moment': [1, 2]}
        mask       = array([[False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, Fals...se, False,  True, False,
        False, False,  True, False,  True, False, False, False,  True,
         True, False]])
        masks      = [array([[False, False, False, False, False, False, False, False, False,
        False, False, False, False, False, Fal...e, False,  True, False,
        False, False,  True, False,  True, False, False, False,  True,
         True, False]])]
        n_outputs  = 1
        n_samples  = 1
        nan_policy = 'propagate'
        paired     = False
        res1da     = array([nan])
        res1db     = (array([nan, nan]),)
        res1dc     = (array([nan, nan]),)
        rng        = Generator(PCG64) at 0x7F2E853FD8C0
        sample     = array([[0.63696169, 0.26978671, 0.04097352, 0.01652764, 0.81327024,
        0.91275558, 0.60663578, 0.72949656, 0.5436...,        nan, 0.13509651,        nan, 0.52535432,
        0.31024188, 0.48583536,        nan,        nan, 0.3577952 ]])
        unpacker   = <function <lambda> at 0x7f2e8598ea70>
______________________________________________________ test_axis_nan_policy_axis_is_None[all_nans-propagate-moment-args12-kwds12-1-1-False-<lambda>] ______________________________________________________
[gw4] linux -- Python 3.10.4 /home/tirthasheshpatel/tensor_libs/virtualenvs/scipy-dev/bin/python
scipy/stats/tests/test_axis_nan_policy.py:366: in test_axis_nan_policy_axis_is_None
    assert_equal(res1db, res1da)
E   AssertionError: 
E   Arrays are not equal
E   
E   (shapes (1, 2), (1,) mismatch)
E    x: array([[nan, nan]])
E    y: array([nan])
        args       = ()
        data       = [array([[nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan, nan, nan],
       [nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
        nan, nan, nan, nan, nan, nan, nan]])]
        data_generator = 'all_nans'
        data_raveled = [array([nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
       nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan, nan,
       nan])]
        ea_str     = None
        eb_str     = None
        ec_str     = None
        hypotest   = <function moment at 0x7f1a9da7ac20>
        kwds       = {'moment': [1, 2]}
        n_outputs  = 1
        n_samples  = 1
        nan_policy = 'propagate'
        paired     = False
        res1da     = array([nan])
        res1db     = (array([nan, nan]),)
        res1dc     = (array([nan, nan]),)
        rng        = Generator(PCG64) at 0x7F1A9D225700
        unpacker   = <function <lambda> at 0x7f1a9d7b6a70>
========================================================================================= short test summary info =========================================================================================
FAILED scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-propagate-moment-args12-kwds12-1-1-False-<lambda>] - ValueError: setting an array element with a sequence.
FAILED scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-omit-moment-args12-kwds12-1-1-False-<lambda>] - ValueError: setting an array element with a sequence.
FAILED scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_fast[mixed-1-raise-moment-args12-kwds12-1-1-False-<lambda>] - ValueError: setting an array element with a sequence.
FAILED scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_axis_is_None[mixed-propagate-moment-args12-kwds12-1-1-False-<lambda>] - AssertionError: 
FAILED scipy/stats/tests/test_axis_nan_policy.py::test_axis_nan_policy_axis_is_None[all_nans-propagate-moment-args12-kwds12-1-1-False-<lambda>] - AssertionError: 
====================================================================================== 5 failed, 268 passed in 4.18s ======================================================================================
```


</details>

Reply:
Try `None` as the last element of the tuple. (That is what it is for all the other functions that have two outputs. I don't remember exactly how it v works, though.)

Reply:
`(stats.moment, tuple(), dict(moment=[1, 2]), 1, 2, False, None)` Works without any failures.

```
ninja: Entering directory `build'
[2/2] Generating scipy/generate-config with a custom command
Build OK
Installing, see meson-install.log...
Installation OK
Running tests for scipy version:1.9.0.dev0+1979.39c9eb2, installed at:/home/tirthasheshpatel/tensor_libs/scipy/build-install/lib/python3.10/site-packages/scipy
=========================================================================================== test session starts ===========================================================================================
platform linux -- Python 3.10.4, pytest-7.0.1, pluggy-1.0.0
rootdir: /home/tirthasheshpatel/tensor_libs/scipy, configfile: pytest.ini
plugins: cov-3.0.0, xdist-2.5.0, forked-1.4.0
gw0 [273] / gw1 [273] / gw2 [273] / gw3 [273] / gw4 [273] / gw5 [273] / gw6 [273] / gw7 [273]
................................................................................................................................................................................................... [ 71%]
..............................................................................                                                                                                                      [100%]
=========================================================================================== 273 passed in 4.10s ===========================================================================================

```

Reply:
Please push. I do think it belongs there.

Reply:
Done!

Reply:
Great! Let me take a final pass, but I think we can get this in today.

Reply:
Slow tests didn't run locally. One of them is failing but the result shape is correct. I have pushed a small change that should resolve the failing test.